src/test/java/com/easypost/Fixture.java

```diff
@@ -431,4 +431,60 @@ public static Map<String, Object> rmaFormOptions() {
 
         return params;
     }
+
+    /**
+     * Basic shipment with different to-from addresses that is reusable in all test coverage.
+     *
+     * @return A map that has the shipment info filled.
+     */
+    public static Map<String, Object> basicCarbonOffsetShipment() {
+        Map<String, Object> basicShipment = new HashMap<>();
+
+        basicShipment.put("to_address", pickupAddress());
+        basicShipment.put("from_address", basicAddress());
+        basicShipment.put("parcel", basicParcel());
+
+        return basicShipment;
+    }
+
+    /**
+     * Shipment that has all detail and different to-from addresses that is reusable in all test coverage.
+     *
+     * @return A map that has the shipment info filled in detail.
+     */
+    public static Map<String, Object> fullCarbonOffsetShipment() {
+        Map<String, Object> fullShipment = new HashMap<>();
+        Map<String, Object> options = new HashMap<>();
+
+        options.put("label_format", "PNG"); // Must be PNG so we can convert to ZPL later.
+        options.put("invoice_number", "123");
+        fullShipment.put("to_address", pickupAddress());
+        fullShipment.put("from_address", basicAddress());
+        fullShipment.put("parcel", basicParcel());
+        fullShipment.put("customs_info", basicCustomsInfo());
+        fullShipment.put("reference", "123");
+        fullShipment.put("options", options);
+
+        return fullShipment;
+    }
+
+    /**
+     * Shipment with one call buy and different to-from addresses that is reusable in all test coverage.
+     *
+     * @return A map that has the shipment info filled for one call buy.
+     */
+    public static Map<String, Object> oneCallBuyCarbonOffsetShipment() {
+        Map<String, Object> oneCallBuyShipment = new HashMap<>();
+        List<Object> carrierAccounts = new ArrayList<>();
+
+        carrierAccounts.add(uspsCarrierAccountID());
+        oneCallBuyShipment.put("to_address", pickupAddress());
+        oneCallBuyShipment.put("from_address", basicAddress());
+        oneCallBuyShipment.put("parcel", basicParcel());
+        oneCallBuyShipment.put("service", uspsService());
+        oneCallBuyShipment.put("carrier_accounts", carrierAccounts);
+        oneCallBuyShipment.put("carrier", usps());
+
+        return oneCallBuyShipment;
+    }
```

Comment:
Issue: Please remove all of these `carbon offset shipment` fixtures as they are identical to the already existing shipment fixtures with a new name. Simply call the `basic`, `full`, or `one-call-buy` shipment fixtures in your tests instead. The point of fixtures is to provide as few data points as possible that can be reused as much as possible and mixed/matched as needed for test data.

Reply:
These are different, slightly. Normal shipment fixtures have the same `to` and `from` address (the `BasicAddress` fixture). For carbon offset to work, the `to` and `from` addresses need to be different. So these shipments have the `BasicAddress` as the `from` address, and the `PickupAddress` as the `to` address.

I'm not a huge fan of this either. Please advise.

Reply:
Interesting. I wondered when this would come to bite us. My suggestion instead is to simply add another basic address fixture so that these are always different. I can handle that separately next sprint when I consolidate all the fixture data. This is fine for now, but it'll get changed in the future.